### PR TITLE
add MsbuildPath option

### PR DESCRIPTION
**Pull request by zhili1208**

### CODE REVIEW

File: src/NuGet.Clients/NuGet.CommandLine/NuGetCommand.resx [hunk 1]
```diff
@@ -5330,4 +5330,7 @@ nuget locals global-packages -list</value>
   <data name="CommandDirectDownload" xml:space="preserve">
     <value>Download directly without populating any caches with metadata or binaries.</value>
   </data>
+  <data name="CommandMSBuildPath" xml:space="preserve">
+    <value>Msbuild Path</value>
```
Comment (joelverhagen) [inline src/NuGet.Clients/NuGet.CommandLine/NuGetCommand.resx]: We should enhance this description as well as mentioning this takes precedence over `MSBuildVersion`.

**Comment by drewgillies** [comment]
:shipit:

File: src/NuGet.Clients/NuGet.CommandLine/Commands/PackCommand.cs [hunk 2]
```diff
@@ -75,6 +75,9 @@ public ICollection<string> Exclude
         [Option(typeof(NuGetCommand), "CommandMSBuildVersion")]
         public string MSBuildVersion { get; set; }
```
Comment (joelverhagen) [inline src/NuGet.Clients/NuGet.CommandLine/Commands/PackCommand.cs:76]: Can we warn when the user passes both `MSBuildVersion` and `MSBuildPath` stating that the `MSBuildPath` is preferred.

File: src/NuGet.Clients/NuGet.CommandLine/Commands/PackCommand.cs [hunk 3]
```diff
@@ -75,6 +75,9 @@ public ICollection<string> Exclude
...
 
+        [Option(typeof(NuGetCommand), "CommandMSBuildPath")]
+        public string MSBuildPath { get; set; }
```
Comment (joelverhagen) [inline src/NuGet.Clients/NuGet.CommandLine/Commands/PackCommand.cs:79]: We need a couple tests for this in NuGet.CommandLine.Test:
1. Verify that `MSBuildPath` is used.
2. Verify that `MSBuildPath` is preferred over `MSBuildVersion`.
3. Verify what happens when `MSBuildPath` is pointing to a file that does not exist.

**Review by joelverhagen — CHANGES_REQUESTED** [review]

**Comment by rohit21agrawal** [comment]
Pack3 is a dotnet cli command , so no this won't have any impact on pack3.

**Comment by jainaashish** [comment]
Add tests as mentioned by Joel, then :shipit:

**Pull request closed without merging**